src/drivers/power_monitor/ina226/ina226.cpp

```diff
@@ -325,6 +162,32 @@ INA226::init()
 	start();
 	_sensor_ok = true;
 
+	_initialized = ret == OK;
+	return ret;
+}
+
+int
+INA226::force_init()
+{
+	_battery.updateBatteryStatus(
+		hrt_absolute_time(),
+		0.0,
+		0.0,
+		false,
+		false, // TODO: selected source?
+		0,
+		0.0,
+		false,
+		false
+	);
+	_battery.publish();
+
+	int ret = init();
+
+	if (ret != OK) {
+		start();
```

Comment:
Huh? When the `init` is not ok, you `start`?

Reply:
I just realized my mistake there. Instead of `start()` I should use `ScheduleDelayed()` directly. 

If `init` fails, it schedules to retry the init again in the future (see the end of `Run()`). I think when I wrote this I was under the impression that a call to `start()` was somehow necessary here, but looking again, it's clearly not.